src/common/rabbitmq/rabbitmq.service.ts

```diff
@@ -0,0 +1,120 @@
+import amqp, { Connection, Channel } from "amqplib/callback_api";
+import {
+  Injectable,
+  Logger,
+  OnModuleDestroy,
+  OnApplicationShutdown,
+} from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+
+/**
+ * Service for publishing messages to a RabbitMQ queue.
+ */
+@Injectable()
+export class RabbitMQService implements OnModuleDestroy, OnApplicationShutdown {
+  private connectionOptions: amqp.Options.Connect;
+  private connection: Connection;
+  private channel: Channel;
+
+  constructor(private readonly configService: ConfigService) {
+    Logger.log("Initializing RabbitMQService.", "RabbitMQService");
+
+    const hostname = this.configService.get<string>("rabbitMq.hostname");
+    const port = this.configService.get<number>("rabbitMq.port");
+    const username = this.configService.get<string>("rabbitMq.username");
+    const password = this.configService.get<string>("rabbitMq.password");
+
+    if (!hostname || !port || !username || !password) {
+      Logger.error(
+        "RabbitMQ is enabled but missing one or more config variables.",
+        "RabbitMQService",
+      );
```

Comment:
`onApplicationBootstrap()` under the [app.module.ts ](https://github.com/SciCatProject/scicat-backend-next/blob/master/src/app.module.ts) tries to connect to the rabbitMQ using the same variables, so there will be two instances of rabbitMQ connections once enabled.
That's some legacy code from the ESS that is no longer used but not removed. Would you prefer me to create new PR to remove that or just delete them from this PR?